astropy/coordinates/baseframe.py

```diff
@@ -600,71 +600,51 @@ def _infer_repr_info(cls, repr_info):
             if isinstance(cls_or_name, str):
                 # TODO: this provides a layer of backwards compatibility in
                 # case the key is a string, but now we want explicit classes.
-                _cls = _get_repr_cls(cls_or_name)
-                repr_info[_cls] = repr_info.pop(cls_or_name)
+                repr_info[_get_repr_cls(cls_or_name)] = repr_info.pop(cls_or_name)
 
         # The default spherical names are 'lon' and 'lat'
-        repr_info.setdefault(
+        sph_repr = repr_info.setdefault(
             r.SphericalRepresentation,
             [RepresentationMapping("lon", "lon"), RepresentationMapping("lat", "lat")],
         )
 
-        sph_component_map = {
```

Comment:
Alternatively, keep this map and just define `lon` and `lat` from it (which, after a bit of looking I do like; clearer!). I.e.,
```
sph_component_map = {m.reprname: m.framename for m in sph_repr}
lon = sph_component_map["lon"]
lat = sph_component_map["lat"]
```

Reply:
Yes, this is shorter.